fastlane/actions/get_latest_github_release_within_same_major.rb

```diff
@@ -0,0 +1,99 @@
+module Fastlane
+  module Actions
+
+    class GetLatestGithubReleaseWithinSameMajorAction < Action
+      def self.run(params)
+        repo_name = params[:repo_name]
+        UI.message("Getting latest release for #{repo_name} on GitHub within same major as the current version")
+
+        GithubApiAction.run(
+          server_url: "https://api.github.com",
+          api_token: ENV["GITHUB_API_TOKEN"],
+          api_bearer: nil,
+          http_method: 'GET',
+          path: "repos/RevenueCat/#{repo_name}/releases",
+          error_handlers: {
+            404 => proc do |result|
+              UI.error("Repository #{repo_name} cannot be found, please double check its name and that you provided a valid API token (if it's a private repository).")
+            end,
+            401 => proc do |result|
+              UI.error("You are not authorized to access #{repo_name}, please make sure you provided a valid API token.")
+            end,
+            '*' => proc do |result|
+              UI.error("GitHub responded with #{result[:status]}:#{result[:body]}")
+            end
+          }
+        ) do |result|
+          current_version = Gem::Version.new(params[:current_version])
+          json = result[:json]
+          if json.count == 0
+            UI.user_error!("Couldn't find latest GitHub release for #{params[:repo_name]}")
+            return
+          end
+
+          highest_version = json.reject { |item| item["prerelease"] }
+            .map { |item| item_version(item) }
+            .select { |item_version| same_major_range?(item_version, current_version) }
+            .max
+
+          UI.message("Version #{highest_version.version} is latest release in the same major live on GitHub.com 🚁")
+          return highest_version.version
+        end
+      end
+
+      private_class_method def self.item_version(item)
```

Comment:
I think this is the correct way of defining a private method but I am not sure lol

Reply:
Yeah this is fine, this works for "static" methods such as these. For normal methods, you need to add them after a `private` delimiter (http://ruby-for-beginners.rubymonstas.org/advanced/private_methods.html). In any case, this is correct 👍